src/function/scalar/string/like.cpp

```diff
@@ -1,13 +1,70 @@
 #include "duckdb/common/exception.hpp"
 #include "duckdb/common/vector_operations/vector_operations.hpp"
 #include "duckdb/function/scalar/string_functions.hpp"
+#include "utf8proc.hpp"
 
 using namespace std;
 
 namespace duckdb {
 
 static bool like_operator(const char *s, const char *pattern, const char *escape);
 
+template <bool IS_UPPER>
+static std::unique_ptr<string_t> to_upper_case(Vector &result, const char *input_data, idx_t input_length) {
```

Comment:
my idea was:
1. pattern to uppercase
2. create `ilike_operator`, where checked string iteratively symbol by symbol to uppercase and check with previously uppercased pattern.

If we don't do p.1, each iteration in `ExecuteFlat` we need to uppercase pattern symbol by symbol.

Reply:
Your idea sounds good to me, but I believe you would be better off not using the BinaryExecutor directly. The BinaryExecutor is intended for simple loops, complex functions should be explicitly handled elsewhere.

This exact problem is encountered in the `regexp_match` function, where we do not want to compile a constant regex pattern for every string. This is solved by using the `bind` function and defining a function that has a switch on whether or not the pattern is constant. Please look at `regexp.cpp` lines 46-93.

I believe this approach will work better for you, and will even be faster because the uppercasing of the pattern will happen exactly once, rather than once per vector processed.